crates/starknet/src/starknet/mod.rs

```diff
@@ -0,0 +1,450 @@
+use std::collections::HashMap;
+use std::time::SystemTime;
+
+use starknet_api::block::{BlockNumber, BlockStatus, BlockTimestamp, GasPrice};
+use starknet_in_rust::definitions::block_context::{
+    BlockContext, StarknetChainId, StarknetOsConfig,
+};
+use starknet_in_rust::definitions::constants::{
+    DEFAULT_CAIRO_RESOURCE_FEE_WEIGHTS, DEFAULT_CONTRACT_STORAGE_COMMITMENT_TREE_HEIGHT,
+    DEFAULT_GLOBAL_STATE_COMMITMENT_TREE_HEIGHT, DEFAULT_INVOKE_TX_MAX_N_STEPS,
+    DEFAULT_VALIDATE_MAX_N_STEPS,
+};
+use starknet_in_rust::execution::TransactionExecutionInfo;
+use starknet_in_rust::state::in_memory_state_reader::InMemoryStateReader;
+use starknet_in_rust::state::BlockInfo;
+use starknet_in_rust::testing::TEST_SEQUENCER_ADDRESS;
+use starknet_in_rust::SierraContractClass;
+use starknet_rs_core::types::{BlockId, TransactionStatus};
+use starknet_types::contract_address::ContractAddress;
+use starknet_types::felt::{ClassHash, Felt, TransactionHash};
+use starknet_types::traits::HashProducer;
+use tracing::error;
+
+use crate::account::Account;
+use crate::blocks::{StarknetBlock, StarknetBlocks};
+use crate::constants::{CAIRO_0_ACCOUNT_CONTRACT_PATH, ERC20_CONTRACT_ADDRESS};
+use crate::error::{Error, Result};
+use crate::predeployed_accounts::PredeployedAccounts;
+use crate::state::StarknetState;
+use crate::traits::{AccountGenerator, Accounted, HashIdentifiedMut, StateChanger};
+use crate::transactions::declare_transaction::DeclareTransactionV1;
+use crate::transactions::declare_transaction_v2::DeclareTransactionV2;
+use crate::transactions::{StarknetTransaction, StarknetTransactions, Transaction};
+use crate::utils;
+
+mod add_declare_transaction;
+mod predeployed;
+
+#[derive(Debug)]
+pub struct StarknetConfig {
+    pub seed: u32,
+    pub total_accounts: u8,
+    pub predeployed_accounts_initial_balance: Felt,
+    pub host: String,
+    pub port: u16,
+    pub timeout: u16,
+    pub gas_price: u64,
+    pub chain_id: StarknetChainId,
+}
+
+impl Default for StarknetConfig {
+    fn default() -> Self {
+        Self {
+            seed: u32::default(),
+            total_accounts: u8::default(),
+            predeployed_accounts_initial_balance: Felt::default(),
+            host: String::default(),
+            port: u16::default(),
+            timeout: u16::default(),
+            gas_price: u64::default(),
+            chain_id: StarknetChainId::TestNet,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct Starknet {
+    pub(in crate::starknet) state: StarknetState,
```

Comment:
there is no need to be defined as pub(in crate::starknet) it could be pub(crate) its the same

Reply:
Those fields initially were declared private by you. This is not needed anymore? I decided to keep that logic restricting access only to friend functions in `starknet` module

Reply:
Where are the friend functions

Reply:
The fields with pub(in crate::starknet) are accessible only in `starknet` module not crate so those fields aren't accessible in `crate::account` module or any other module

Reply:
`friend` is more like c++ notation that is mimicked with `pub(in some::module)`

Reply:
okay, that was misunderstanding about the pub(in crate::starknet).